utilities/additionalProperties.go

```diff
@@ -0,0 +1,34 @@
+package utilities
+
+import "encoding/json"
+
+// MapAdditionalProperties append additional properties to destination struct map
+func MapAdditionalProperties(destinationMap additionalProperties, sourceMap additionalProperties) {
+	destinationMap.appendMap(sourceMap)
+}
+
+// UnmarshalAdditionalProperties unmarshals additional properties and remove fields that exists on parent struct
+func UnmarshalAdditionalProperties(input []byte, keysToRemove ...string) (map[string]any, error) {
+	var destinationMap additionalProperties
+	err := destinationMap.unmarshalAdditionalProperties(input, keysToRemove)
+	return destinationMap, err
+}
+
+// additionalProperties helper struct for handling additional properties in models
+type additionalProperties map[string]any
+
+func (dstMap *additionalProperties) appendMap(srcMap additionalProperties) {
```

Comment:
we an remove this struct of additonalProperties, because we are implementing Marshallar and unmarshaller for additionalProperties